substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -3375,6 +3426,8 @@ impl<T: Config> Pallet<T> {
 
 	fn do_claim_commission(who: T::AccountId, pool_id: PoolId) -> DispatchResult {
 		let bonded_pool = BondedPool::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?;
+		// ensure pool is not in an un-migrated state.
+		ensure!(!Self::api_pool_needs_delegate_migration(pool_id), Error::<T>::NotMigrated);
```

Comment:
Same here -- should we keep all these checks at the extrinsic level? In this case, the line above needs to be moved there too.